server/src/test/java/org/elasticsearch/indices/store/IndicesStoreIntegrationIT.java

```diff
@@ -480,13 +480,19 @@ private Path shardDirectory(String server, Index index, int shard) {
         return paths[0];
     }
 
-    private boolean waitForShardDeletion(final String server, final Index index, final int shard) throws InterruptedException {
-        awaitBusy(() -> !Files.exists(shardDirectory(server, index, shard)));
+    private boolean waitForShardDeletion(final String server, final Index index, final int shard) throws Exception {
+        try {
+            assertBusy(() -> assertFalse(Files.exists(shardDirectory(server, index, shard))));
+        } catch (AssertionError ignore) {
+        }
         return Files.exists(shardDirectory(server, index, shard));
     }
 
-    private boolean waitForIndexDeletion(final String server, final Index index) throws InterruptedException {
-        awaitBusy(() -> !Files.exists(indexDirectory(server, index)));
+    private boolean waitForIndexDeletion(final String server, final Index index) throws Exception {
+        try {
+            assertBusy(() -> assertFalse(Files.exists(indexDirectory(server, index))));
+        } catch (AssertionError ignore) {
+        }
```

Comment:
I wonder if we could adapt this test so that we don't the empty catch blocks? If we don't always assert on the same result, maybe we can pass in a consumer that asserts on the result and we would not need to return the boolean anymore?

Reply:
I have no idea about how to remove the catch blocks.  WDYT?